### PR TITLE
python3Packages.resolvelib: 0.5.5 -> 0.8.1

### PR DESCRIPTION
**Pull request by cpcloud**
###### Motivation for this change

Bump resolvelib

I'm not entirely sure how to test ansible.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cpcloud** [comment]
@FRidh It looks like this will not be compatible with `ansible-galaxy` per https://github.com/ansible/ansible/blob/devel/requirements.txt.

Should I go ahead an split this into two separate packages (`resolvelib-0_5` for ansible, and plain `resolvelib` for future dependents, `pdm` being the first one))

**Comment by FRidh** [comment]
in `python-packages.nix` we do not allow multiple versions of the same package that could end up in the runtime closure. So, if pdm is used as a library (which is why it would go in `python-packages.nix`) there is no option but having one broken package.

**Comment by cpcloud** [comment]
Does the story change if `pdm` is an application? I think that'll be the primary way it's used.

**Comment by FRidh** [comment]
Actually ansible is already a library, so then it is fine to add a second version of the library. Preferably put the older version it in the ansible folder.

If pdm is packaged as application, then yes, it can have other dependency versions.

**Comment by cpcloud** [comment]
@FRidh I split `resolvelib` into 2 packages, and I'll reopen the pdm PR against `master` as an application.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -8493,7 +8493,18 @@ in {
 
   resampy = callPackage ../development/python-modules/resampy { };
 
-  resolvelib = callPackage ../development/python-modules/resolvelib { };
+  resolvelib-0_5 = callPackage ../development/python-modules/resolvelib {
```
Comment (risicle) [inline pkgs/top-level/python-packages.nix:8496]: The way we normally name attributes this would be `resolvelib_0_5`

Comment (risicle) [reply pkgs/top-level/python-packages.nix:8496]: But also having multiple versions of a single package violates the `python-packages.nix` policy.

Comment (cpcloud) [reply pkgs/top-level/python-packages.nix:8496]: Right , but there's only Ansible and pdm (once the PR lands) that use it and they are both applications. 

Alternatively we'd have to mark Ansible broken or fix it upstream to land pdm.

I can move the Ansible version into the Ansible package if that would help.

Comment (mweinelt) [reply pkgs/top-level/python-packages.nix:8496]: Ansible is a library, since it is consumed by 

- ansible-runner
- ansible-lint
- pytest-ansible

Comment (cpcloud) [reply pkgs/top-level/python-packages.nix:8496]: @fridh does it make sense to move the newer version of resolvelib into the pdm directory for now?

Comment (mweinelt) [reply pkgs/top-level/python-packages.nix:8496]: You can use `packageOverrides` to override the resolvelib version in `python3Packages` if you're using `buildPythonApplication`.

**Comment by cpcloud** [comment]
Closing, will incorporate this directly into #155390

**Pull request closed without merging**